### PR TITLE
Cleanup pointless CUDA_LAMBDA guards

### PR DESCRIPTION
**Pull request by dalg24**
Also dropping CUDA_LAMBDA and CUDA_LDG_INTRINSIC from the information printed in`Cuda::print_configuration()`

### CODE REVIEW

**Review by masterleinad — APPROVED** [review]

**Comment by dalg24** [comment]
https://github.com/kokkos/kokkos/blob/f1faaf8398041c47a4eceba7a28634729da39513/core/unit_test/TestViewMapping_a.hpp#L1012-L1031
fails
```
1: [ RUN      ] openmp.view_mapping
1: /var/jenkins/workspace/Kokkos_PR-7388/core/unit_test/TestViewMapping_a.hpp:1031: Failure
1: Expected equality of these values:
1:   errors
1:     Which is: 30
1:   0
1: [  FAILED  ] openmp.view_mapping (59 ms)
4: [ RUN      ] cuda.view_mapping
4: /var/jenkins/workspace/Kokkos_PR-7388/core/unit_test/TestViewMapping_a.hpp:1031: Failure
4: Expected equality of these values:
4:   errors
4:     Which is: 30
4:   0
4: [  FAILED  ] cuda.view_mapping (59 ms)
```

**Comment by dalg24** [comment]
Interestingly the Clang+CUDA build passed

File: core/unit_test/TestViewMapping_a.hpp [hunk 1]
```diff
@@ -1009,9 +1009,6 @@ void test_view_mapping() {
     ASSERT_EQ(a.use_count(), 1);
     ASSERT_EQ(b.use_count(), 0);
 
-#if !defined(KOKKOS_ENABLE_CUDA) || !defined(KOKKOS_ENABLE_CUDA_LAMBDA)
-    // Cannot launch host lambda when CUDA lambda is enabled.
```
Comment (dalg24) [inline core/unit_test/TestViewMapping_a.hpp:1013]: https://github.com/kokkos/kokkos/pull/5628#discussion_r1094515352
Daniel will need to reexplain to me...

Comment (masterleinad) [reply core/unit_test/TestViewMapping_a.hpp:1013]: Something funky with reference counting going on with lambdas. It's always `7` here. It works if you turn it into a functor...
```diff
diff --git a/core/unit_test/TestViewMapping_a.hpp b/core/unit_test/TestViewMapping_a.hpp
index 49edc16f8..83da1b7e2 100644
--- a/core/unit_test/TestViewMapping_a.hpp
+++ b/core/unit_test/TestViewMapping_a.hpp
@@ -24,6 +24,25 @@
 
 namespace Test {
 
+       template <typename Space>
+       struct DummyFunctor {
+
+Kokkos::View<int *, Space> a;
+
+               KOKKOS_FUNCTION void operator()(int i, int& e) const {
+         // for parallel_reduce we copy the functor into a combined
+          // functor-reducer object (with reference-counting on) before
+          // constructing the ParallelReduce object (with reference-counting
+          // turned off). When the parallel dispatch accepts a move for the
+          // lambda, this count should become 2.
+
+          if (a.use_count() != 3) { Kokkos::printf("1: %d should be 3\n", a.use_count()); ++e; }
+         Kokkos::View<int*, Space> x = a;
+          if (a.use_count() != 3) { Kokkos::printf("2: %d should be 3\n", a.use_count()); ++e; }
+          if (x.use_count() != 3) { Kokkos::printf("3: %d should be 3\n", a.use_count()); ++e; }
+               }
+       };
+
 template <class Space>
 void test_view_mapping() {
   using ExecSpace = typename Space::execution_space;
@@ -1014,7 +1033,7 @@ void test_view_mapping() {
 
     int errors = 0;
     Kokkos::parallel_reduce(
-        Kokkos::RangePolicy<host_exec_space>(0, 10),
+        Kokkos::RangePolicy<host_exec_space>(0, 10), DummyFunctor<Space>{a}, /*
         KOKKOS_LAMBDA(int, int& e) {
           // for parallel_reduce we copy the functor into a combined
           // functor-reducer object (with reference-counting on) before
@@ -1022,11 +1041,11 @@ void test_view_mapping() {
           // turned off). When the parallel dispatch accepts a move for the
           // lambda, this count should become 2.
 
-          if (a.use_count() != 3) ++e;
+          if (a.use_count() != 3) { Kokkos::printf("1: %d should be 3\n", a.use_count()); ++e; }
           V x = a;
-          if (a.use_count() != 3) ++e;
-          if (x.use_count() != 3) ++e;
-        },
+          if (a.use_count() != 3) { Kokkos::printf("2: %d should be 3\n", a.use_count()); ++e; }
+          if (x.use_count() != 3) { Kokkos::printf("3: %d should be 3\n", a.use_count()); ++e; }
+        },*/
         errors);
     ASSERT_EQ(errors, 0);
   }
```

File: core/unit_test/TestViewMapping_a.hpp [hunk 2]
```diff
@@ -1009,9 +1009,7 @@ void test_view_mapping() {
...
-
+#if defined(KOKKOS_ENABLE_CUDA) && defined(KOKKOS_COMPILER_NVCC)
```
Comment (masterleinad) [inline core/unit_test/TestViewMapping_a.hpp:1012]: Let's at least provide a comment explaining that the use case is higher (but constant) and that we see the expected numbers if we were to turn the lambda into a functor.

Comment (dalg24) [reply core/unit_test/TestViewMapping_a.hpp:1012]: Please take care of it I am swamped

Comment (masterleinad) [reply core/unit_test/TestViewMapping_a.hpp:1014]: fb12f3f

**Review by Rombur — APPROVED** [review]

**Comment by crtrott** [comment]
Just to comment here: the reason is that a __host__ __device__ lambda under the the hood in nvcc creates some wrapper objects possibly even two versions of the lambda (one for use in host functions one in device functions). All of that makes it messy and potentially subject to the CUDA version.

**Review by crtrott — APPROVED** [review]
Ok the condition looks good now :-)

**Comment by dalg24** [comment]
Unrelated failure in one of the CUDA builds (GPU not detected at runtime)

**Pull request merged**